projectboard/main.go

```diff
@@ -0,0 +1,200 @@
+/*
+Copyright © 2023 The Kubernetes Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"time"
+
+	"github.com/google/go-github/v48/github"
+	githubql "github.com/shurcooL/githubv4"
+	"golang.org/x/oauth2"
+)
+
+const (
+	// perPage is the number of items to return per page.
+	perPage = 100
+	// tokenEnv is the environment variable that contains the
+	// GitHub token to use for authentication.
+	// GITHUB_TOKEN is a personal access token with the following scopes:
+	// - repo (all)
+	// - read:org
+	// - project (all)
+	tokenEnv = "GITHUB_TOKEN"
+	// orgNameEnv is the environment variable that contains the
+	// name of the GitHub organization to query.
+	// e.g. kubernetes
+	orgNameEnv = "GITHUB_ORG"
+	// projectNameEnv is the environment variable that contains the
+	// name of the GitHub project to query.
+	// e.g. SIG Auth
+	projectNameEnv = "GITHUB_PROJECT"
```

Comment:
Let's use the GitHub  `Project number` instead of `Project Name`?

Multiple GH projects can have the same name.